src/llm/http_llm_calculator.cc

```diff
@@ -0,0 +1,523 @@
+//*****************************************************************************
+// Copyright 2024 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <algorithm>
+#include <string>
+#include <thread>
+#include <chrono>
+#include <ctime>
+#include <unordered_map>
+#include <optional>
+#include <sstream>
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#include "mediapipe/framework/calculator_framework.h"
+#include "mediapipe/framework/port/canonical_errors.h"
+#pragma GCC diagnostic pop
+
+#include <openvino/openvino.hpp>
+#include <continuous_batching_pipeline.hpp>
+
+#include <rapidjson/stringbuffer.h>
+#include <rapidjson/writer.h>
+
+#include "llmnoderesources.hpp"
+#include "http_payload.hpp"
+
+using namespace rapidjson;
+
+namespace mediapipe {
+
+using chat_entry_t = std::unordered_map<std::string, std::string>;
+using chat_t = std::vector<chat_entry_t>;
+
+class OpenAIChatCompletionsRequest {
```

Comment:
I think we should move it outside of this file in case we potentially want to use this calculator with other kinds of requests.

Reply:
Can we move it outside later, once we need it?